kinto/core/resource/__init__.py

```diff
@@ -95,6 +95,26 @@ def register_service(endpoint_type, settings):
             argument_getter = getattr(viewset, '%s_arguments' % endpoint_type)
             view_args = argument_getter(resource_cls, method)
 
+            # FIXME: We need to support JSON-merge validation, so the approch
+            # is to validate the Content-Type as a JSON variant and then cast
+            # the request as a regular JSON request.
+            def validate_json_content_type(req, *args, **kargs):
+                """
+                Cast JSON Content-Type variations as JSON for validation as
+                they are JSON anyways.
+                """
+
+                if re.match('^application/(.*?)json$', req.content_type):
+                    req.validated_content_type = req.content_type
+                    req.content_type = 'application/json'
+                else:
+                    req.validated_content_type = None
+
+                return req
+
+            if 'validators' in view_args:
+                view_args['validators'].insert(0, validate_json_content_type)
```

Comment:
We can probably move this to `core.resource.viewset` (and handle a global refactor later with Kinto/kinto#880)

Or, better, contribute the fix to Cornice: https://github.com/Cornices/cornice/blob/master/cornice/validators/__init__.py#L36
(i.e. `re.match()` on the content_type header instead of equality)